DrivewayDentDeletion/Operators/prereqs.sh

```diff
@@ -76,90 +99,78 @@ oc adm policy add-scc-to-group privileged system:serviceaccounts:${test_namespac
-export DOCKER_REGISTRY="image-registry.openshift-image-registry.svc:5000"
-export username=image-bot
-kubectl -n ${dev_namespace} create serviceaccount image-bot
-oc -n ${dev_namespace} policy add-role-to-user registry-editor system:serviceaccount:${dev_namespace}:image-bot
+echo "INFO: Adding permission for '$dev_namespace' to write images to openshift local registry in the '$test_namespace'"
 # enable dev namespace to push to test namespace
 oc -n ${test_namespace} policy add-role-to-user registry-editor system:serviceaccount:${dev_namespace}:image-bot
-export password="$(oc -n ${dev_namespace} serviceaccounts get-token image-bot)"
-
-echo -e "\nCreating secrets to push images to openshift local registry"
-oc create -n ${dev_namespace} secret docker-registry cicd-${dev_namespace} --docker-server=${DOCKER_REGISTRY} \
-  --docker-username=${username} --docker-password=${password} -o yaml | oc apply -f -
 
 echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
 
-# Creating a new secret as the type of entitlement key is 'kubernetes.io/DOCKERCONFIGJSON_ER' but we need secret of type 'kubernetes.io/basic-auth'
-# to pull imags from the ER
-echo "Creating secret to pull base images from Entitled Registry"
-cat << EOF | oc apply --namespace ${dev_namespace} -f -
+declare -a image_projects=("${dev_namespace}" "${test_namespace}")
+
+for image_project in "${image_projects[@]}" #for_outer
+do
+  echo "INFO: Generating user, database name and password for the postgres database in the '$image_project' namespace"
+  DB_POD=$(oc get pod -n $POSTGRES_NAMESPACE -l name=postgresql -o jsonpath='{.items[].metadata.name}')
+  DB_USER=$(echo ${image_project}_${SUFFIX} | sed 's/-/_/g')
+  DB_NAME="db_$DB_USER"
+  DB_PASS=$(LC_ALL=C tr -dc 'A-Za-z0-9' </dev/urandom | head -c 32 ; echo)
+  PASSWORD_ENCODED=$(echo -n ${DB_PASS} | base64)
+
+  echo "INFO: Creating a secret for the database user '$DB_USER' in the database '$DB_NAME' with the password generated"
+  # everything inside 'data' must be in the base64 encoded form
+  cat << EOF | oc apply -f -
 apiVersion: v1
 kind: Secret
 metadata:
-  name: er-pull-secret
-  annotations:
-    tekton.dev/docker-0: ${ER_REGISTRY}
-type: kubernetes.io/basic-auth
+  namespace: $image_project
+  name: postgres-credential
+type: Opaque
 stringData:
-  username: ${ER_USERNAME}
-  password: ${ER_PASSWORD}
+  username: $DB_USER
+data:
+  password: ${PASSWORD_ENCODED}
 EOF
 
-echo -e "\n----------------------------------------------------------------------------------------------------------------------------------------------------------\n"
-
-echo "Waiting for postgres to be ready"
-oc wait -n postgres --for=condition=available deploymentconfig --timeout=20m postgresql
+  echo -e "INFO: Configuring postgres in the '$image_project' namespace with the user '$DB_USER' and database name '$DB_NAME' and suffix '$SUFFIX'\n"
```

Comment:
The `$image_project` doesn't really matter here, all it's doing is creating a database in the postgres instance that could be accessed from anywhere. Also not sure the suffix is relevant. Suggest changing to:
```suggestion
  echo -e "INFO: Creating '$DB_NAME' database and '$DB_USER' user in the postgres instance in the ${POSTGRES_NAMESPACE} namespace\n"
```